pkg/container/client.go

```diff
@@ -157,6 +157,15 @@ func (client dockerClient) GetContainer(containerID t.ContainerID) (Container, e
 		return Container{}, err
 	}
 
+	containerNetworkMode := strings.Split(string(containerInfo.HostConfig.NetworkMode), ":")
+	if len(containerNetworkMode) == 2 {
+		parentContainer, err := client.api.ContainerInspect(bg, containerNetworkMode[1])
+		if err != nil {
+			log.Debug("Unable to fetch parentContainer.")
+		}
+		containerInfo.HostConfig.NetworkMode = container.NetworkMode(fmt.Sprintf("container:%s", parentContainer.Name))
```

Comment:
```suggestion
		} else {
			containerInfo.HostConfig.NetworkMode = container.NetworkMode(fmt.Sprintf("container:%s", parentContainer.Name))
		}
```

If we don't get a parentContainer we shouldn't be updating the `NetworkMode`. In fact, the container should probably be skipped with an error if this happens, but at least this way will just fall back to the old (current) behaviour of watchtower.